cli/cmd/compose/up.go

```diff
@@ -117,6 +122,8 @@ func upCommand(p *projectOptions, contextType string) *cobra.Command {
 		flags.BoolVar(&opts.noStart, "no-start", false, "Don't start the services after creating them.")
 		flags.BoolVar(&opts.cascadeStop, "abort-on-container-exit", false, "Stops all containers if any container was stopped. Incompatible with -d")
 		flags.StringVar(&opts.exitCodeFrom, "exit-code-from", "", "Return the exit code of the selected service container. Implies --abort-on-container-exit")
+		flags.IntVarP(&opts.timeout, "timeout", "t", 10, "Use this timeout in seconds for container shutdown when attached or when containers are already running.")
```

Comment:
IMO `time.Duration` would be better here. So you can let the user use `1m`, `1s` and so on.

Reply:
this aligns with Docker CLI and other places where we defined timeout
I'd prefer we focus on docker-compose parity before we debate UX improvements :-P